src/models/Order.js

```diff
@@ -115,6 +117,30 @@ OrderSchema.pre('findOne', async function (next) {
   console.log('Pre-findOne finished');
   next();
 });
+
+// Middleware to send order completion email when the order status changes to "complete"
+OrderSchema.post('save', async function (doc) {
+  try {
+    if (doc.orderStatus === 'complete') {
+      const user = await User.findById(doc.user); // Assuming you have a User model
```

Comment:
Since we have the user field in the order schema, we can get this info by calling `populate` instead of a separate user query (see next comment)